sympy/printing/lambdarepr.py

```diff
@@ -69,6 +69,59 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+class NumPyPrinter(LambdaPrinter):
+    """
+    Numpy printer which handles vectorized piecewise functions,
+    logical operators, etc.
+    """
+    _default_settings = {
+        "order": "none",
+        "full_prec": "auto",
+    }
+
+    def _print_seq(self, seq, delimiter=', '):
+        "General sequence printer: converts to tuple"
+        # Print tuples here instead of lists because numba supports
+        #     tuples in nopython mode.
+        return '({},)'.format(delimiter.join(self._print(item) for item in seq))
+
+    def _print_MatMul(self, expr):
+        "Matrix multiplication printer"
+        return '({0})'.format(').dot('.join(self._print(i) for i in expr.args))
+
+    def _print_Piecewise(self, expr):
+        "Piecewise function printer"
+        # Print tuples here instead of lists because numba may add support
+        #     for select in nopython mode; see numba#1313 on github.
+        exprs = '({0},)'.format(','.join(self._print(arg.expr) for arg in expr.args))
+        conds = '({0},)'.format(','.join(self._print(arg.cond) for arg in expr.args))
+        # If (default_value, True) is a (expr, cond) tuple in a Piecewise object
+        #     it will behave the same as passing the 'default' kwarg to select()
+        #     *as long as* it is the last element in expr.args.
+        # If this is not the case, it may be triggered prematurely.
+        return 'select({0}, {1})'.format(conds, exprs)
```

Comment:
The SymPy default for Piecewise is `nan`, whereas you told me that select defaults to 0. Should we make this match SymPy?